atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -350,13 +399,16 @@ private static CassandraKeyValueService createAndInitialize(
             CassandraKeyValueServiceConfig config,
             Supplier<CassandraKeyValueServiceRuntimeConfig> runtimeConfigSupplier,
             CassandraClientPool clientPool,
+            Optional<AsyncClusterSession> asyncSession,
             CassandraMutationTimestampProvider mutationTimestampProvider,
             Logger log,
             boolean initializeAsync) {
+        asyncSession.ifPresent(AsyncClusterSession::start);
```

Comment:
what does it mean for the session to start here?

`AsyncSessionManager` creates sessions, one per cluster combination, yet if you have multiple transaction managers/things that use the the same async session, you will still call `start` multiple times, and you will then have `n` starts

if you have to do this every time that you get given a session it might make sense to just have the session manager "start" the session for you, or better still, have a static factory inside `AsyncSessionManagerImpl` with a method like `createAndStartHealthChecks` which gives you back a "started" `AsyncSessionManager`.

Reply:
I agree, the test I used did not create multiple transaction managers/things and I left it like that. Having this in mind I now changed `finalize` method for this class.